reactive/rerunner.go

```diff
@@ -369,7 +369,9 @@ func (r *Rerunner) run() {
 		time.Sleep(WriteThenReadDelay)
 	}
 	r.cache.cleanInvalidated()
-	ctx := context.WithValue(r.ctx, cacheKey{}, r.cache)
+	ctx, cancel := context.WithCancel(r.ctx)
```

Comment:
Please add a brief comment that describes why this is here.

Reply:
Done

Reply:
Thanks!
Grammar: I don't know how to read the first sentence. Between "context we" I am unable to follow the connection.
How about, "Cancel the context passed to "run". Canceling the context ensures that libraries let go of the context when they might otherwise hold onto it long past this run."